Include route names in kong api requests

kalilsn opened the pull request:
https://github.com/department-of-veterans-affairs/vets-contrib/issues/3287

We've had route names defined in our config files, but they haven't made it into the Kong database because of how kongfig handled them in the past. This is a preliminary change to making kongfig stop using route ids, which will ensure that these route names make it to the kong database. After this is merged, I intend to `kongfig apply` our config against all environments before moving on to relying on route names alone (and ultimately removing the IDs from our config).

Commit:
Include route names in kong api requests

Since kong 1.0 introduced route names, kongfig's custom route name
handling doesn't play nicely with the official attribute. This commit
lets kongfig use the new route names while still allowing the old format
to work.

dvogel commented on src/readKongApi.js:
```diff
@@ -106,10 +106,15 @@ export function parseRoute({ id, created_at, updated_at, service, ...rest }, ser
         }
     }
-    return { name: (name || id), id, attributes: {...rest}, _info: { id, updated_at, created_at } };
+    return { name, id, attributes: {...rest}, _info: { id, updated_at, created_at } };
```
From the comment above:

> Kong 1.0 added names as an actual property of a route, so it should be specified as attributes.name ...

So should name be next to `...rest`, inside `attributes`?

kalilsn replied:
Turns out this was not a wise choice 😅. Adding name to attributes here prevented kongfig from detecting when the name in kong didn't match the name in the config.

Commit:
Ensure attributes.name is always set

kalilsn commented:
Evidently I didn't commit the tests I wrote before 😫

Commit:
Add some tests for route name changes

Commit:
Merge branch 'master' into kalilsn/route-ids

Commit:
Increment version

jasalisbury commented on src/actions.js:
```diff
@@ -42,7 +42,7 @@ export function addServiceRoute(serviceId, { name, attributes }) {
         endpoint: {name: 'routes', params: {serviceId, routeName: name}},
         method: 'POST',
-        body: assign({}, attributes, { service: { id: serviceId } })
+        body: { name, ...attributes, service: { id: serviceId } }
```
Do we need to explicitly add name here since name is in `attributes`? Is this to keep backwards compatibility with Kong < 1.0?

kalilsn replied:
We might not, since we're ensuring it's in attributes in `parseRoute`. Unfortunately, I'm not 100% confident that the route argument to these action functions always comes from the output of `parseRoute`. So this might be unnecessary, but I don't think it does any harm to use the top level name in case it's not in attributes.

kalilsn replied:
Turns out this was not a wise choice 😅. Adding name to attributes here prevented kongfig from detecting when the name in kong didn't match the name in the config.

maddy-jo commented on test/routes.test.js:
```diff
@@ -0,0 +1,71 @@
+    }]
+};
+
+const serviceName = 'mockbin'
```
move before `baseConfig` and use as `name`?

Commit:
Merge branch 'master' into kalilsn/route-ids

Commit:
Revert "Ensure attributes.name is always set"

This reverts commit b58511dfbedbe38c3ae8d3ff3597072e10b05603.

Commit:
Add integration test for route names

Commit:
Remove useless tests

Commit:
Formatting

Commit:
Don't print standard logging output in test

kalilsn commented:
I fixed the issue preventing kong from detecting changes, and replaced my tests with an integration test that actually tests the behavior I'm looking for (kongfig adding route names to existing routes that don't have them yet).

Commit:
Merge branch 'master' into kalilsn/route-ids

Commit:
2.1.4
